polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1583,43 +1580,194 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
         CGM.getContext().getTargetAddressSpace(PointeeType.getAddressSpace()));
   }
 
-  if (T->isBuiltinType() || isa<clang::EnumType>(T)) {
-    if (T->isBooleanType()) {
-      OpBuilder Builder(TheModule->getContext());
-      return Builder.getIntegerType(8);
-    }
-
-    llvm::Type *Ty = CGM.getTypes().ConvertType(QualType(T, 0));
+  if (isa<clang::EnumType>(T)) {
```

Comment:
is it possible for an EnumType to be a BooleanType/VoidTy/FloatTy...?

Reply:
by spec it must be an integer type

Reply:
Correct, a plain enum type is an integer type.